### DISCUSSION TITLE
Feature requests ivm "Evaluatie"

### DISCUSSION THREAD

**Discussion by IngeVerbeek** [dodona-edu/feedback-and-support#10] [Feature requests]
Ik zou het handig vinden om een **overzicht van alle evaluaties** te hebben als leerkracht. Welke evalutaties heb ik al afgerond, welke evaluaties zijn al vrijgegeven, ... Nu moet ik elke afzonderlijke reeks aanklikken om te weten wat ik al geëvalueerd heb en welke evaluaties ik al heb vrijgegeven.

**Comment by bmesuere** [comment]
Bedankt voor je suggestie. Los van het overzicht zou het ook op de cursuspagina zelf duidelijker moeten zijn welke reeksen een evaluatie hebben (zowel voor lesgevers als studenten).

**Reply by IngeVerbeek** [reply]
Inderdaad.
Ik stel vast dat mijn leerlingen amper zien dat er feedback gegeven wordt en ze dus ook niet lezen :(

Ik zou het zelfs interessant vinden om te kunnen zien of ze de feedback hebben gelezen (allé ja, opengeklikt) of niet.

**Comment by pdawyndt** [comment]
Zou ook handig zijn om op de pagina van een student in de cursus een overzicht te krijgen van de resultaten van alle evaluaties waaraan die student heeft deelgenomen. Dan zie je in één oogopslag hoe die student in de cursus gepresteerd heeft. Een dergelijk overzicht zou dan voor de student ook handig zijn om in één keer al zijn eigen punten voor alle evaluaties te zien.

**Reply by IngeVerbeek** [reply]
Klopt! En als hij dan ook kan doorklikken naar de feedback (en zelf eventueel zien welke feedback hij nog niet gelezen heeft), dan wordt die misschien meer gelezen.

Ik merk dat de meeste leerlingen nu amper weten dat er feedback gegeven wordt en hem niet lezen. Bovendien klikken ze automatisch op de titel van de opgave: dat brengt hen naar de opgave en niet naar hun oplossing/feedback. Daarvoor moeten ze op dat kleine pijltje helemaal rechts klikken, maar dat doen ze niet vanzelf ...

**Comment by IngeVerbeek** [comment]
Een ander idee rond evaluaties:

Deadlines voor individuele leerlingen of voor groepen van leerlingen met een bepaald label kunnen aanpassen voor een hele reeks, zodat ik het niet voor elke individuele oefening moet aanpassen. Dat is handig als er meerdere klassen in dezelfde dodona-cursus zitten, als bepaalde leerlingen meer tijd krijgen voor het indienen van hun oefeningen owv ziekte, ...

EN/OF

Voor één oefeningenreeks meerdere evaluaties kunnen aanmaken. Dan kan ik na een les een evaluatie aanmaken en vrijgeven voor bvb 10 studenten, en de week nadien voor de 2 overige studenten een nieuwe evaluatie aanmaken (met een nieuwe deadline).

**Reply by bmesuere** [reply]
Dit is een stuk moeilijker om toe te voegen omdat de ingediende oplossingen niet aan reeksen vast hangen, maar gewoon aan de cursus. Het zou dus een (heel) grote aanpassing vragen.

**Reply by IngeVerbeek** [reply]
Zou het mogelijk zijn om een reeks gemakkelijk te dupliceren? Dan zou ik een reeks kunnen dupliceren en aan elke reeks een andere deadline geven en via de geheime link de reeks bezorgen aan de juiste studenten. 
Als ik dit nu wil doen moet ik een nieuwe reeks toevoegen en dan handmatig alle oefeningen opnieuw toevoegen.
Dit zou het bovenstaande probleem al een beetje oplossen.

**Reply by BTWS2** [reply]
@IngeVerbeek voor het kopiëren van een reeks is er een issue aangemaakt (https://github.com/dodona-edu/dodona/issues/2809).

**Comment by IngeVerbeek** [comment]
Ik denk dat ik een bug gevonden heb:
- ik heb een evaluatie aangemaakt, alles van feedback voorzien en vrijgegeven.
- ik heb vervolgens enkele leerlingen gevraagd om hun code aan te passen ahv de gegeven feedback 
- ik heb daarna de evaluatie aangepast en bij deze leerlingen op "Selecteer een andere ingediende oplossing" geklikt.
- ik heb ook die ingediende oplossing nagekeken en van feedback voorzien. (De score heb ik niet aangepast.)
Nu blijven die nieuw geselecteerde oplossingen in mijn lijstje staan van evaluaties die ik nog moet afronden. Mijn balkje "evaluatie voortgang" staat nu niet op "voltooid" en zelfs als ik de oplossingen opnieuw bekijk lukt het me niet om de evaluatie te voltooien voor deze leerlingen.

**Reply by bmesuere** [reply]
Bedankt voor het melden van dit probleem. Waarschijnlijk wordt ze opgelost met de aanpassingen uit #3405 die we volgende week zullen meenemen in een nieuwe versie van Dodona.
